File: CS/CustomActionControl.Module.Win/ActionItems/CustomBarActionItemsFactoryProvider.cs
using DevExpress.ExpressApp.Win.Templates;
using DevExpress.XtraBars;

namespace CustomActionControl.Module.Win.ActionItems {
    public class CustomBarActionItemsFactoryProvider : IBarActionItemsFactoryProvider {
        public BarActionItemsFactory CreateBarActionItemsFactory(BarManager barManager) {
            return new CustomBarActionItemsFactory(barManager);
        }
    }
}

File: VB/CustomActionControl.Module.Win/ActionItems/CustomBarActionItemsFactoryProvider.vb
Imports DevExpress.ExpressApp.Win.Templates
Imports DevExpress.XtraBars

Namespace CustomActionControl.Module.Win.ActionItems
    Public Class CustomBarActionItemsFactoryProvider
        Implements IBarActionItemsFactoryProvider

        Public Function CreateBarActionItemsFactory(barManager As BarManager) As BarActionItemsFactory Implements IBarActionItemsFactoryProvider.CreateBarActionItemsFactory
            Return New CustomBarActionItemsFactory(barManager)
        End Function
    End Class
End Namespace
